lib/Star/Kata/Model/Objective/ConfigurableObjective.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\Model\Objective;
+
+/**
+ * Class ConfigurableObjective
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\Model\Objective
+ */
+final class ConfigurableObjective implements Objective
+{
+    /**
+     * @var string
+     */
+    private $definition;
+
+    /**
+     * @var \Closure
+     */
+    private $assertCode;
+
+    /**
+     * @param string   $definition
+     * @param callable $assertCode
+     */
+    public function __construct($definition, \Closure $assertCode)
+    {
+        $this->definition = $definition;
+        $this->assertCode = $assertCode;
+    }
+
+    /**
+     * Validate the Objective.
+     *
+     * @return ObjectiveResult
+     */
+    public function validate()
+    {
+        $closure = $this->assertCode;
+        $closure();
+    }
+
+    /**
+     * The name of the objective.
+     *
+     * @return string
+     */
+    public function getName()
```

Comment:
`getName` will need to be renamed to `getDefinition` or `getDescription`.

@octopixelle If this pr is merge in master, you could probably change the objective's name to objective description or definition.

@kevinarcher 

Which options would you see best fit:

* Objective::getDefintion()
* Objective::getDescription()
* other ?

This field would contain the full information about the goal of the objective, to be rendered in the console, or GUI.